tests/test_webhook.py

```diff
@@ -0,0 +1,170 @@
+import json
+import unittest
+
+import unittest.mock as mock
+
+from boddle import boddle
+
+import webhook
+
+TEST_ACCESS_TOKEN = 'TEST_ACCESS_TOKEN'
+TEST_VERIFY_TOKEN = 'TEST_VERIFY_TOKEN'
+TEST_API_KEY = 'TEST_API_KEY'
+
+
+class TestWebhook(unittest.TestCase):
+    def setUp(self):
+        settings_patch = mock.patch('webhook.msbot.settings')
+        self.addCleanup(settings_patch.stop)
+        self.settings_mock = settings_patch.start()
+        self.settings_mock.configure_mock(
+            **{
+                'PAGE_ACCESS_TOKEN': TEST_ACCESS_TOKEN,
+                'VERIFY_TOKEN': TEST_VERIFY_TOKEN,
+                'API_KEY': TEST_API_KEY,
+            }
+        )
+
+        requests_patch = mock.patch('webhook.requests')
+        self.addCleanup(requests_patch.stop)
+        self.requests_mock = requests_patch.start()
+
+    def test_send_message(self):
+        mock_send_psid = 123456
+        mock_message = 'Hello'
+        webhook.send_message(mock_send_psid, mock_message)
+
+        self.requests_mock.post.assert_called_once()
+
+        # Test request body
+        _, request_body = self.requests_mock.post.call_args
+        self.assertEqual(
```

Comment:
Should this just be one big assertEqual with a constructed request_body? Like, rather than doing all these dictionary lookups in the call arguments, I could just test that the entire request_body matches some static dictionary I define?

Reply:
Not sure what the best way to go about it is, argument is that I'm pretty much already embedding the structure of the object in the way I've written the test, so might as well reduce the number of asserts

Reply:
does python have a way to do deep nested asserts? if so than that'd be nicer but this isn't that big of a deal.

Reply:
Implemented changes!